cpio/cpio.go

```diff
@@ -146,12 +146,34 @@ func (cr *countingReader) Seek(offset int64, whence int) (int64, error) {
 	if offset == 0 {
 		return cr.curPos, nil
 	}
-	b := make([]byte, offset)
-	n, err := io.ReadFull(cr, b)
-	if err != nil && err != io.EOF {
-		return 0, err
+	totalRead := int64(0)
+	const chunkSize = int64(1024 * 1024)
+	buf := make([]byte, chunkSize)
+
+	remaining := offset
+	for remaining > 0 {
+		// Define chunk size to read
+		toRead := chunkSize
+		if remaining < chunkSize {
+			toRead = remaining
+		}
+
+		n, err := cr.Read(buf[:toRead])
+		totalRead += int64(n)
+		remaining -= int64(n)
+		if err != nil && err != io.EOF {
+			// If all was read, skip error
+			if totalRead >= offset {
+				err = nil
+			}
+			return 0, err
+		}
+
+		if err == io.EOF {
+			break
+		}
```

Comment:
Hi, thanks for your contribution! This is definitely worth fixing.

I believe this can be addressed much more simply by using `io.Discard`:
```suggestion
	return io.CopyN(io.Discard, cr, offset)
```

It may be a tiny bit less efficient but the simplicity is worth it in my opinion.